src/openlcb/BroadcastTimeDefs.cxx

```diff
@@ -0,0 +1,161 @@
+/** @copyright
+ * Copyright (c) 2018, Stuart W. Baker
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are  permitted provided that the following conditions are met:
+ *
+ *  - Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  - Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @file BroadcastTimeDefs.cxx
+ *
+ * Static definitions for implementations of the OpenLCB Broadcast Time
+ * Protocol.
+ *
+ * @author Stuart W. Baker
+ * @date 30 July 2022
+ */
+
+#include "BroadcastTimeDefs.hxx"
+
+#include <string>
+
+namespace openlcb
+{
+
+extern "C"
+{
+// normally requires _GNU_SOURCE
```

Comment:
when this happens in a cxx file, you can add `#define _WHATEVER_SOURCE` to the top of that cxx file before you include any headers. Then it's guaranteed that you will get the respective declaration.
Sometimes this runs into problems with duplicate definitions of preprocessor symbols. That needs to be fixed in the other occurrence of the `#define`, which is invariably in a hxx file, by adding a wrapper of `#ifndef _WHATEVER_SOURCE  #define _WHATEVER_SOURCE #endif`